modules/SofaGeneralAnimationLoop/MechanicalMatrixMapper.inl

```diff
@@ -0,0 +1,518 @@
+* your option) any later version.                                             *
+*                                                                             *
+* This library is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this library; if not, write to the Free Software Foundation,     *
+* Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA.          *
+*******************************************************************************
+*                               SOFA :: Modules                               *
+*                                                                             *
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#ifndef SOFA_COMPONENT_FORCEFIELD_MechanicalMatrixMapper_INL
+#define SOFA_COMPONENT_FORCEFIELD_MechanicalMatrixMapper_INL
+
+#include "MechanicalMatrixMapper.h"
+#include <sofa/core/visual/VisualParams.h>
+#include <sofa/helper/gl/template.h>
+#include <sofa/helper/system/config.h>
+#include <sofa/helper/rmath.h>
+
+// accumulate jacobian
+#include <sofa/core/ExecParams.h>
+#include <sofa/core/objectmodel/BaseContext.h>
+#include <sofa/core/behavior/MechanicalState.h>
+#include <sofa/defaulttype/MapMapSparseMatrix.h>
+
+// verify timing
+#include <sofa/helper/system/thread/CTime.h>
+
+//  Eigen Sparse Matrix
+#include <Eigen/Sparse>
+
+namespace sofa
+{
+
+namespace component
+{
+
+namespace interactionforcefield
+{
+
+template<class DataTypes1, class DataTypes2>
+MechanicalMatrixMapper<DataTypes1, DataTypes2>::MechanicalMatrixMapper()
+    :
+      d_forceFieldList(initData(&d_forceFieldList,"forceFieldList","ForceFields to work on (by default will take all)")),
+      l_nodeToParse(initLink("nodeToParse","link to the node on which the component will work")),
+      l_mechanicalState(initLink("mechanicalState","The mechanicalState with which the component will work on")),
+      l_mappedMass(initLink("mass","mass with which the component will work on")),
+      l_forceField(initLink("forceField","The ForceField(s) attached to this node"))
+{
+}
+
+template<class DataTypes1, class DataTypes2>
+void MechanicalMatrixMapper<DataTypes1, DataTypes2>::init()
+{
+    sofa::core::behavior::BaseInteractionForceField::init();
+
+    if (mstate1.get() == NULL || mstate2.get() == NULL)
```

Comment:
you are testing it well here. But there are several access to the mstate1 and 2 in the code. How will the scene behave if mstate are not found?
maybe you could use the paremeter: ComponentState::Invalid;

Reply:
Thanks for the remarks, added in commit https://github.com/sofa-framework/sofa/pull/721/commits/451aefec7860c994b747b667fc97073c3afefc52